wordpress-mobile/WordPress-Android
# Issue/12319 new discover endpoint load data

Parent issue #12319

This is a follow up for https://github.com/wordpress-mobile/WordPress-Android/pull/12517

Merge instructions:

1. Review this PR
2. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/12517 is merged
3. Update target branch to issue/12319-new-discover-endpoint-main-issue
4. Remove "Not Ready for Merge" tag
5. Merge this PR


This PR connects the UI with the discover /cards service. Certain parts of the implementation aren't final - eg. subscription to the `FetchDiscoverCardsEnded`. Since we are merging this into a feature branch it shouldn't be an issue. I'll keep coming with follow up PRs.


To test:
1. Clear app data
2. Turn on the FF
3. Open Reader
4. Select topics and click on done
5. Notice the discover feed contains some posts - they are not related to the topics you picked yet -> I simply hardcoded "android" as a default topic for now.

PR submission checklist:
